internal/mode/static/state/graph/graph.go

```diff
@@ -112,6 +119,54 @@ func (g *Graph) IsReferenced(resourceType client.Object, nsname types.Namespaced
 	}
 }
 
+// IsNGFPolicyRelevant returns whether the NGF Policy is a part of the Graph, or targets a resource in the Graph.
+func (g *Graph) IsNGFPolicyRelevant(
+	policy policies.Policy,
+	gvk schema.GroupVersionKind,
+	nsname types.NamespacedName,
+) bool {
+	key := PolicyKey{
+		NsName: nsname,
+		GVK:    gvk,
+	}
+
+	if _, exists := g.NGFPolicies[key]; exists {
+		return true
+	}
+
+	if policy == nil {
```

Comment:
Isn't obj nil on a delete event?

Reply:
> Isn't obj nil on a delete event?

pol, ok := obj.(policies.Policy) 

this will panic in obj is nil, right?

Reply:
oh, it will not panic, because in that case, obj will be objType.

this means `pol` cannot be nil, but it will be just an "empty" policy on delete. I wonder if logic in IsNGFPolicyRelevant needs to be updated?

Reply:
I think the logic is correct

Reply:
On a delete event the object isn't nil (I was wrong about that), just empty. So we should panic here like you suggested, but otherwise the logic works